c2corg_images/scripts/migrate.py

```diff
@@ -1,45 +1,89 @@
-from c2corg_images.scripts import MultithreadProcessor
-from c2corg_images.resizing import original_pattern, create_resized_images, resized_keys
-from c2corg_images.storage import v5_storage, temp_storage, active_storage
+import os
 import requests
+from sqlalchemy import create_engine
+from c2corg_images.scripts import MultithreadProcessor
+from c2corg_images.resizing import create_resized_images, resized_keys
+from c2corg_images.storage import temp_storage, active_storage
 
 import logging
 log = logging.getLogger(__name__)
 
+V5_DATABASE_URL = os.environ.get('V5_DATABASE_URL', None)
+
 
 class Migrator(MultithreadProcessor):
 
+    def __init__(self, *args, **kwargs):
+        super(MultithreadProcessor, self).__init__(*args, **kwargs)
+        self.v5_engine = create_engine(V5_DATABASE_URL)
+
     def do_keys(self):
-        for key in v5_storage.keys():
-            match = original_pattern.match(key)
-            if match:
-                yield key
+        batch_size = 500
+        offset = 0
+        connection = self.v5_engine.connect()
+        sql = """
+SELECT count(*) AS count
+FROM app_images_archives;
+"""
+        count_result = connection.execute(sql)
+        for row in count_result:
+            total = row['count']
+
+        while offset < total:
+            sql = """
+SELECT filename
+FROM app_images_archives
+LIMIT {} OFFSET {};
+""".format(batch_size, offset)
+
+            result = connection.execute(sql)
+            for row in result:
+                yield row['filename']
+
+            offset += batch_size
+        connection.close()
 
     def do_process_key(self, key):
-        if not self.force and active_storage.exists(key):
-            with self.lock:
-                self.skipped += 1
-            return
+        to_create = [key] + resized_keys(key)
+        if not self.force:
+            for key_to_create in list(to_create):
+                if active_storage.exists(key_to_create):
+                    to_create.remove(key_to_create)
+            if len(to_create) == 0:
+                log.debug('{} skipping'.format(key))
+                with self.lock:
+                    self.skipped += 1
+                    return
 
         log.debug('{} getting file in temp storage'.format(key))
-        import os
-        if os.environ['STORAGE_BACKEND'] == 's3':
-            r = requests.get('http://s.camptocamp.org/uploads/images/{}'.
-                             format(key),
-                             stream=True)
-            with open(temp_storage.object_path(key), 'wb') as fd:
-                for chunk in r.iter_content(None):
-                    fd.write(chunk)
-        else:
-            v5_storage.copy(key, temp_storage)
+        tries = 3
+        success = False
+        while tries > 0 and not success:
+            try:
+                r = requests.get('http://s.camptocamp.org/uploads/images/{}'.
```

Comment:
It probably makes sense to set a connection timeout: http://docs.python-requests.org/en/master/user/quickstart/#timeouts